src/Nethermind/Nethermind.JsonRpc/Modules/RpcModuleProvider.cs

```diff
@@ -120,29 +122,36 @@ public void Return(string methodName, IModule module)
             _pools[result.ModuleType].ReturnModule(module);
         }
 
-        private IDictionary<string, (MethodInfo, bool)> GetMethodDict(Type type)
+        private IDictionary<string, (MethodInfo, bool, RpcEndpoint)> GetMethodDict(Type type)
         {
             var methods = type.GetMethods(BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly);
             return methods.ToDictionary(
                 x => x.Name.Trim(),
-                x => (x, x.GetCustomAttribute<JsonRpcMethodAttribute>()?.IsSharable ?? true));
+                x =>
+                {
+                    JsonRpcMethodAttribute? jsonRpcMethodAttribute = x.GetCustomAttribute<JsonRpcMethodAttribute>();
+                    return (x, jsonRpcMethodAttribute?.IsSharable ?? true, jsonRpcMethodAttribute?.Availability ?? RpcEndpoint.All);
+                });
```

Comment:
Maybe instead of returning tuple (MethodInfo, bool, RpcEndpoint) return (MethodInfo, JsonRpcMethodAttribute) ?

Or better just MethodInfo and get JsonRpcMethodAttribute later when needed?

Reply:
class/struct JsonRpcMethodInfo { MethodInfo, sharable, supoportedEndpoints }